urllib3/response.py

```diff
@@ -327,9 +389,19 @@ def read(self, amt=None, decode_content=None, cache_content=False):
                     # no harm in redundantly calling close.
                     self._fp.close()
                     flush_decoder = True
+                    if (self.enforce_content_length and
+                            self.length_remaining not in (0, None)):
```

Comment:
This is still kinda counter to urllib3's style. I would break the second statement out into a method (`self.has_remaining_data`, perhaps?) so we can fit everything on one line easily without supplemental parentheses. Also, as @sigmavirus24 mentioned, we go with a 100-character line here, rather than the 79-character line recommended by PEP-8, so I'm not sure if you even need to do any refactoring to fit it into one line.